.github/workflows/compile-macos-arm64.yaml

```
@@ -0,0 +1,35 @@
+name: macOS(ARM64)
+
+on: [pull_request]
+
+jobs:
+  compile-macOS-ARM64:
+    runs-on: macos-latest
+    steps:
+    - name: Check Platform
+      run: |
+        if [ ! $(uname -m) = "arm64" ]; then
+            echo "Not on ARM64" >&2
+            exit 1
+        fi
+    - name: Setup macOS
+      run: |
+        # Copied from mpi4py/mpi-publish
+        # create gfortran symlink
+        cd $(brew --prefix)/bin
+        gfortran=$(ls gfortran-* | sort | head -n 1)
+        sudo ln -s $gfortran gfortran
+        # install autotools
+        brew install autoconf
+        brew install automake
+        brew install libtool
+        # unlink libevent
+        brew unlink libevent || true
+    - uses: actions/checkout@v4
+      with:
+            submodules: recursive
+    - name: Build Open MPI
+      run: |
+        ./autogen.pl
+        ./configure
+        make -j $(nproc)
```

@wenduwan @jsquyres  Sorry, guys, I messed up. This is macOS, not Linux! 
This is the proper way to do it on macOS:

```suggestion
        make -j $(sysctl -n hw.logicalcpu)
```